grails-app/migrations/views/cycle-count-session.sql

```diff
@@ -26,7 +27,9 @@ CREATE OR REPLACE VIEW cycle_count_session AS (
 	JOIN location on product_availability.location_id = location.id
 	JOIN product on product_availability.product_id = product.id
 	JOIN inventory on location.inventory_id = inventory.id
+    LEFT OUTER JOIN cycle_count_request ON product.id = cycle_count_request.product_id AND location.id = cycle_count_request.facility_id
     WHERE product_availability.quantity_on_hand > 0
-	GROUP BY location.id, product.id, abc_class
+       AND (cycle_count_request.id IS NULL OR (cycle_count_request.status <> 'COMPLETED' AND cycle_count_request.status <> 'CANCELED'))
```

Comment:
basically we have to join the cycle count request table, but since we won't delete the completed and canceled requests, we need to filter them out in order to have unique result per product/facility pair.
We have a constraint for product/facility pair to be unique, but only if a request is not completed or canceled.
Since the candidates don't need to look into completed and canceled requests, I filtered them out.

Reply:
Does the "cycle_count_request.id IS NULL" work if there isn't a cycle count request? I'd assume it'd fail if cycle_count_request is null.

Reply:
hmm, why would it fail in your opinion? 
We LEFT JOIN the `cycle_count_request` table, so essentially we take all the rows from the `cycle_count_session` table, and join the `cycle_count_request` even if it doesn't exist (so then all the rows from `cycle_count_request` would be `null`).
So this check basically does this: return all the records from the `cycle_count_session` where the cycle count request doesn't exist or its status is not neither COMPLETED nor CANCELED.

Reply:
I was thinking too Java-based. You're right about the left join